signer/src/proto/convert.rs

```diff
@@ -585,6 +586,96 @@ impl TryFrom<proto::StacksTransactionSignRequest> for StacksTransactionSignReque
     }
 }
 
+impl From<model::EncryptedDkgShares> for proto::sbtc::signer::v1::backups::DkgShares {
+    fn from(value: model::EncryptedDkgShares) -> Self {
+        let signer_set_public_keys = value
+            .signer_set_public_keys
+            .into_iter()
+            .map(proto::PublicKey::from)
+            .collect::<Vec<_>>();
+
+        // Convert DKG status enum
+        let dkg_shares_status = match value.dkg_shares_status {
+            model::DkgSharesStatus::Unverified => {
+                proto::sbtc::signer::v1::backups::DkgSharesStatus::Unverified
+            }
+            model::DkgSharesStatus::Verified => {
+                proto::sbtc::signer::v1::backups::DkgSharesStatus::Verified
+            }
+            model::DkgSharesStatus::Failed => {
+                proto::sbtc::signer::v1::backups::DkgSharesStatus::Failed
+            }
+        };
+
+        proto::sbtc::signer::v1::backups::DkgShares {
+            aggregate_key: Some(value.aggregate_key.into()),
+            tweaked_aggregate_key: Some(value.tweaked_aggregate_key.into()),
+            encrypted_private_shares: value.encrypted_private_shares,
+            public_shares: value.public_shares,
+            script_pubkey: value.script_pubkey.to_bytes(),
+            signer_set_public_keys,
+            signature_share_threshold: value.signature_share_threshold as u32,
+            dkg_shares_status: dkg_shares_status.into(),
+            started_at_bitcoin_block_hash: Some(value.started_at_bitcoin_block_hash.into()),
+            started_at_bitcoin_block_height: *value.started_at_bitcoin_block_height,
+        }
+    }
+}
+
+impl TryFrom<proto::sbtc::signer::v1::backups::DkgShares> for model::EncryptedDkgShares {
+    type Error = Error;
+
+    fn try_from(value: proto::sbtc::signer::v1::backups::DkgShares) -> Result<Self, Self::Error> {
+        let signer_set_public_keys = value
+            .signer_set_public_keys
+            .into_iter()
+            .map(|pk| pk.try_into())
+            .collect::<Result<Vec<_>, _>>()?;
+
+        // Convert DKG status enum
+        let dkg_shares_status =
+            proto::sbtc::signer::v1::backups::DkgSharesStatus::try_from(value.dkg_shares_status)
+                .map_err(|_| Error::TypeConversion)?; // Handle invalid enum value
```

Comment:
There should be a helper function implemented on `DkgShares` to do the conversion. I think it'll be called `value.dkg_shares_status()` or something.